### PR TITLE
bugfix/15729-annotation-drag-selected

### LINKED ISSUE #15729: Stock Tools: After an annotation's drag&drop, annotation should remain selected
#### Expected behaviour
After drag&drop action, the annotation should remain selected (all control points should be visible, the popup should be visible)

#### Actual behaviour
Annotation is deselected. Control points and popup are hidden

#### Live demo with steps to reproduce
[Demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/stock/demo/stock-tools-gui)

Steps:
- draw an annotation
- select the annotation (popup shows up)
- drag&drop annotation
- popup and control points are missing

edit

#### Product
Highcharts Stock

### PR DESCRIPTION
**Pull request by khlieng**
Fixed #15729, annotation did not stay selected after dragging.

### CODE REVIEW

**Review by pawelfus — CHANGES_REQUESTED** [review]
See: https://github.com/highcharts/highcharts/pull/15738#pullrequestreview-662251718

**Comment by khlieng** [comment]
@pawelfus should it also stay selected after dragging a control point?

**Comment by pawelfus** [comment]
> @pawelfus should it also stay selected after dragging a control point?

Yes, it should, nice catch!

File: ts/Extensions/Annotations/NavigationBindings.ts [hunk 1]
```diff
@@ -571,19 +571,27 @@ class NavigationBindings {
...
+                // Element could be removed in the child action, e.g. button
+                (clickEvent.target as any).parentNode &&
+                // TO DO: Polyfill for IE11?
+                !closestPolyfill(clickEvent.target as any, '.' + PREFIX + 'popup')
+            ) {
+                fireEvent(navigation, 'closePopup');
+            } else if (activeAnnotation.cancelClick) {
+                setTimeout((): void => {
+                    activeAnnotation.cancelClick = false;
+                }, 0);
```
Comment (pawelfus) [inline ts/Extensions/Annotations/NavigationBindings.ts:593]: Could you add an explanation (a comment) why we need to use `setTimeout()` here? The idea of this block was to simply close popup and deselect the annotation. Then, if any button is selected, start drawing a new annotation.

**Review by pawelfus — CHANGES_REQUESTED** [review]

**Review by pawelfus — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**